
diff --git a/config/arm64_cn9k_linux_gcc b/config/arm64_cn9k_linux_gcc
index 105ad1d695..f6ea583967 100644
--- a/config/arm64_cn9k_linux_gcc
+++ b/config/arm64_cn9k_linux_gcc
@@ -1,9 +1,10 @@
 [binaries]
-c = ['ccache', 'aarch64-marvell-linux-gnu-gcc']
-cpp = ['ccache', 'aarch64-marvell-linux-gnu-cpp']
-ar = 'aarch64-marvell-linux-gnu-gcc-ar'
-strip = 'aarch64-marvell-linux-gnu-strip'
-pkgconfig = 'pkg-config'
+c = ['ccache', 'aarch64-none-linux-gnu-gcc']
+cpp = ['ccache', 'aarch64-none-linux-gnu-g++']
+ar = 'aarch64-none-linux-gnu-gcc-ar'
+strip = 'aarch64-none-linux-gnu-strip'
+pkgconfig = 'pkg-config'
+pkg-config = 'pkg-config'

 [host_machine]
 system = 'linux'

