--- a/arch/arm/dts/cn9130-crb.dtsi	2026-04-28 16:33:21.000000000 +0800
+++ b/arch/arm/dts/cn9130-crb.dtsi	2026-05-25 18:01:56.408835239 +0800
@@ -70,7 +70,7 @@
 	 */
 		/*   0 1 2 3 4 5 6 7 8 9 */
 	pin-func = < 1 1 1 1 1 1 1 1 1 1
-		     1 3 1 0 0 0 0 0 0 3 >;
+		     1 3 2 0 0 0 0 0 0 3 >;
 };
 
 /* on-board eMMC - U6 */
@@ -127,10 +127,10 @@
 	pin-func = < 3   3   3   3   3   3   3   3   3   3
 		     3   3   0   3   3   3   3   0   0   0
 		     0   0   0   0   0   0   0   0   0   0
-		     0   0   0   6   9   2   2   2   2   0
-		     8   8   8   8   0   0   0   7   0   0xa
-		     0   0xa 9   7   0   0   0xe 0xe 0xe 0xe
-		     0xe 0xe 0>;
+		     0   0   0   0   0   2   2   2   2   0
+		     8   8   7   7   0   0   0   7   0   0
+		     0   0   0   7   0   0   0   0   0   0
+		     0   0   0>;
 
 	cp0_sdhci_cd_pins_crb: cp0-sdhci-cd-pins-crb {
 		marvell,pins = < 55 >;
@@ -148,8 +148,8 @@
 	};
 
 	cp0_xsmi_pins_crb: cp0-xsmi-pins-crb {
-		marvell,pins = < 42 43 >;
-		marvell,function = <8>;
+		marvell,pins = < 42 43 47 53>;
+		marvell,function = <7>;
 	};
 
 };
@@ -162,12 +162,24 @@
 	pinctrl-0 = <&cp0_i2c0_pins>;
 	status = "okay";
 	clock-frequency = <100000>;
+
+	tpm@35 {
+		compatible = "st,st33ktpm2xi2c", "tcg,tpm-tis-i2c";
+		reg = <0x35>;
+	};
+
+	eeprom@53 {
+		compatible = "atmel,24c256";
+		reg = <0x53>;
+		pagesize = <64>;
+	};
 };
 
 &cp0_i2c1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&cp0_i2c1_pins>;
 	status = "okay";
+	clock-frequency = <100000>;
 };
 
 &cp0_sdhci0 {
@@ -229,17 +241,17 @@
 	pinctrl-0 = <&cp0_smi_pins_crb>;
 	status = "okay";
 	phy0: ethernet-phy@0 {
-		reg = <0>;
+		reg = <5>;
 	};
-	switch6: ethernet-switch@6 {
-		reg = <6>;
+	phy1: ethernet-phy@1 {
+		reg = <1>;
 	};
 };
 
 &cp0_xmdio {
 	pinctrl-names = "default";
 	pinctrl-0 = <&cp0_xsmi_pins_crb>;
-	status = "okay";
+	status = "disabled";
 	nbaset_phy0: ethernet-phy@0 {
 		reg = <0>;
 	};
@@ -252,19 +264,22 @@
 &cp0_eth0 {
 	status = "okay";
 	phy-mode = "sfi";
+	local-mac-address = [02 00 00 00 00 01];
 };
 
 &cp0_eth1 {
 	status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&cp0_ge1_rgmii_pins>;
-	phy = <&phy0>;
+	phy = <&phy1>;
 	phy-mode = "rgmii-id";
+	local-mac-address = [02 00 00 00 00 02];
 };
 
 &cp0_eth2 {
 	/* Disable it for now, as mainline does not support this IF yet */
 	status = "okay";
-	phy = <&nbaset_phy0>;
-	phy-mode = "sgmii-2500";
+	phy = <&phy0>;
+	phy-mode = "sgmii";
+	local-mac-address = [02 00 00 00 00 03];
 };

--- a/arch/arm/dts/cn9130-crb-A.dts	2026-04-28 16:33:21.000000000 +0800
+++ b/arch/arm/dts/cn9130-crb-A.dts	2026-05-22 15:45:42.605617207 +0800
@@ -6,7 +6,7 @@
 #include "cn9130-crb.dtsi"
 
 / {
-	model = "CN9130-CRB-A";
+	model = "CN9130-CRB-A-William";
 	compatible = "marvell,cn9130-crb-A",
 		"marvell,cn9130",
 		"marvell,armada-ap806-quad",
